### ISSUE TITLE
librispeech deepspeech validation loss differs between pytorch and jax, but wer aligns

### ISSUE THREAD

**Issue by wyfEmma** [mlcommons/algorithmic-efficiency#924]
*Provide a general description of the issue*
validation loss:
<img width="1800" height="1050" alt="Image" src="https://github.com/user-attachments/assets/f262cf59-5214-497c-8282-b9235e4a8fc5" />

wer:

<img width="2112" height="1504" alt="Image" src="https://github.com/user-attachments/assets/2acc29ac-c075-479e-967a-9f5297da8b37" />

**Comment by wyfEmma** [comment]
this is how jax submission calculates ctc_loss  :

submission_runner calls eval_model https://github.com/mlcommons/algorithmic-efficiency/blob/58491177c1a0cfcf75f6b2331c512a7d6facd842/submission_runner.py#L465

eval_model calls eval_model_on_split for the validation set https://github.com/mlcommons/algorithmic-efficiency/blob/58491177c1a0cfcf75f6b2331c512a7d6facd842/algoperf/spec.py#L347

then it calls eval steps, which calls _eval_step: https://github.com/mlcommons/algorithmic-efficiency/blob/58491177c1a0cfcf75f6b2331c512a7d6facd842/algoperf/workloads/librispeech_conformer/librispeech_jax/workload.py#L362

**Comment by wyfEmma** [comment]
in jax workload, to get the number of examples across all devices, it sums the loss['n_valid_examples'] across all devices.
which means it assumes that loss['n_valid_examples'] represents examples on one GPU and summing num of examples on all devices gives us the number of valid examples on all GPUs https://github.com/mlcommons/algorithmic-efficiency/blob/58491177c1a0cfcf75f6b2331c512a7d6facd842/algoperf/workloads/librispeech_conformer/librispeech_jax/workload.py#L357

**Comment by wyfEmma** [comment]
however, printing the loss['n_valid_examples'], it gives us the shape of 256 

<img width="1670" height="960" alt="Image" src="https://github.com/user-attachments/assets/66205ea2-74d8-4573-adf7-a78112424828" />

, which matches the eval_batch_size of librispeech https://github.com/mlcommons/algorithmic-efficiency/blob/58491177c1a0cfcf75f6b2331c512a7d6facd842/algoperf/workloads/librispeech_conformer/workload.py#L70

this means that the  loss['n_valid_examples'] already hold data across all devices, multiplying it by device_count upscaled the number of examples by 4.

when ctc_loss is calculated, in jax the denominator is 4 times larger, so the loss is 4 times smaller, this explains the difference we see in the ctc_loss

**Comment by wyfEmma** [comment]
I think the reason that loss[n_valid_example] contains the data cross all device is because we are now using jit instead of pmap

the fix: remove the replication by 4 here https://github.com/mlcommons/algorithmic-efficiency/blob/58491177c1a0cfcf75f6b2331c512a7d6facd842/algoperf/workloads/librispeech_conformer/librispeech_jax/workload.py#L357

**Comment by wyfEmma** [comment]
after the fix ctc loss aligns:

step graph with CTC_loss: 

<img width="1800" height="1050" alt="Image" src="https://github.com/user-attachments/assets/eb1ee18e-f6da-4b01-aa46-59eed0456e46" />

time plot:

<img width="1800" height="1050" alt="Image" src="https://github.com/user-attachments/assets/a6e6c05c-1bc4-4af5-8a11-50ccd1c71615" />
